### PR TITLE
error message refinement for flow_execution.py: single node execution

### PR DESCRIPTION
**Pull request by chw-microsoft**
# Description

error message refinement for single node validation errors in flow_execution.py 

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by zzn2 — DISMISSED** [review]

**Review by zzn2 — DISMISSED** [review]

**Review by zzn2 — APPROVED** [review]

File: src/promptflow/promptflow/executor/_errors.py [hunk 1]
```diff
@@ -165,6 +165,10 @@ class NodeOutputNotFound(UserErrorException):
     pass
 
 
+class SingleNodeValidationError(UserErrorException):
+    pass
```
Comment (0mza987) [inline src/promptflow/promptflow/executor/_errors.py:169]: add one line summary?

File: src/promptflow/tests/executor/e2etests/test_executor_validation.py [hunk 2]
```diff
@@ -289,11 +314,12 @@ def test_single_node_input_type_invalid(
         # Single Node run - the inputs are from flow_inputs + dependency_nodes_outputs
         with pytest.raises(error_class) as exe_info:
             FlowExecutor.load_and_exec_node(
-                flow_file=get_yaml_file(flow_folder, path_root),
+                flow_file="flow.dag.yaml",
```
Comment (0mza987) [inline src/promptflow/tests/executor/e2etests/test_executor_validation.py:317]: there should be a common constant iirc, try use that

File: src/promptflow/promptflow/executor/_errors.py (hunk 1 above)
Comment (chw-microsoft) [reply src/promptflow/promptflow/executor/_errors.py:169]: this name is clear and this is non-public

File: src/promptflow/tests/executor/e2etests/test_executor_validation.py (hunk 2 above)
Comment (chw-microsoft) [reply src/promptflow/tests/executor/e2etests/test_executor_validation.py]: good point

**Review by 0mza987 — APPROVED** [review]

**Review by zzn2 — APPROVED** [review]

**Pull request merged**